teps/0041-move-image-entrypoint-lookup-to-the-taskrun-pod.md

```diff
@@ -0,0 +1,161 @@
+---
+status: proposed
+title: Move Image Entrypoint Lookup to the TaskRun Pod
+creation-date: '2021-01-11'
+last-updated: '2021-01-21'
+authors:
+- '@yaoxiaoqi'
+---
+
+# TEP-0041: Move Image Entrypoint Lookup to the TaskRun Pod
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Proposal](#proposal)
+  - [Add Flags to EntryPoint Binary](#add-flags-to-entrypoint-binary)
+  - [Resolve Image in EntryPoint Binary](#resolve-image-in-entrypoint-binary)
+  - [Notes/Caveats](#notescaveats)
+    - [Entrypoint Cache](#entrypoint-cache)
+    - [Error Reporting Delay](#error-reporting-delay)
+    - [Inconsistency](#inconsistency)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes to move the image entrypoint lookup to entrypoint binary that
+runs in the TaskRun pod.
+
+## Motivation
+
+Currently, Tekton controller pod has access to all service accounts and will do
+entry point lookup using `go-containerregistry` for the images that don't have
+any command specified for them. This potentially might cause permission denied
```

Comment:
Not immediately relevant for this proposal, but FWIW I think some of this behavior is buggy. I think this should fail for any spec that doesn't reference an image digest and doesn't specify the command - part of what the entrypoint looking wants to do is guarantee here is a consistent image for all steps in a pod.

Reply:
This also raises an interesting question of whether or not we want/need similar image consistency at the PipelineRun level 🤔 

I don't believe we try and do this today, but theoretically images could modify between pods.

Reply:
@wlynch indeed. I don't think we currently are using the image by-digest in the pod we create, which means images could be different between pods (in some cases). knative does this (using the image by-digest in the resulting pod) but we don't.